MaxText/train.py

```diff
@@ -174,17 +174,19 @@ def train_step(model, config, state, data, dropout_rng):
     data[k] = v[:config.global_batch_size_to_train_on,:]
 
   def loss_fn(params):
+    padding_mask = data['targets_segmentation'] != 0
     logits, intermediate_outputs = model.apply({'params': params},
                          data['inputs'],
                          data['inputs_position'],
                          decoder_segment_ids=data['inputs_segmentation'],
+                         padding_mask=padding_mask,
                          enable_dropout=config.enable_dropout,
                          rngs={'dropout': rng1, 'aqt': aqt_rng}, mutable='intermediates')
     one_hot_targets = jax.nn.one_hot(data['targets'], config.vocab_size)
     xent, _ = max_utils.cross_entropy_with_logits(logits, one_hot_targets, 0.0)
     xent = nn.with_logical_constraint(xent, ('activation_batch', 'activation_length'))
     # Mask out paddings at the end of each example.
-    xent = xent * (data['inputs_segmentation'] != 0)
+    xent = xent * padding_mask
```

Comment:
Sorry what is the different between inputs_segmentation and padding_mask? I'm having trouble following!

Reply:
Oh, there is a jump in this code. 
```
- xent = xent * (data['inputs_segmentation'] != 0)
+ xent = xent * (data['targets_segmentation'] != 0)
```
see [Matt's change](https://github.com/google/maxtext/commit/4f1bbce69df7c0f6ada4a90a43ba9961df732750)

Then

```
- xent = xent * (data['targets_segmentation'] != 0)
+ xent = xent * padding_mask
```

Reply:
Sorry none of this is explaining to me the padding_mask? Don't we just have `data['targets_segmentation']`?

Reply:
Discussed offline. We won't need `padding_mask` any more